CCDB/src/CCDBDownloader.cxx

```diff
@@ -516,12 +516,20 @@ void CCDBDownloader::transferFinished(CURL* easy_handle, CURLcode curlCode)
         } else if (300 <= httpCode && httpCode < 400 && performData->locInd < locations.size()) {
           followRedirect(performData, easy_handle, locations, rescheduled, contentRetrieved);
         } else if (200 <= httpCode && httpCode < 300) {
-          contentRetrieved = true;
+          contentRetrieved = true; // Can be overruled by following timeout check
         }
       } else {
         LOG(error) << loggingMessage;
       }
 
+      // Check for timeout
+      curl_off_t transferTime = 0;
+      curl_easy_getinfo(easy_handle, CURLINFO_TOTAL_TIME_T, &transferTime);
+      if (transferTime / 1000 >= mRequestTimeoutMS) {
```

Comment:
Are you sure this is the best way to check for a timeout? Shouldn't the API give you some kind of error flag that the transfer failed due to a timeout?

Reply:
It turns out there is an error flag set for that - I missed it during my initial search. I'll update the code shortly.